tools/clone-tutorial.py

```diff
@@ -0,0 +1,268 @@
+test_harness_dir = kt_home + '/_data/harnesses'
+orig_name_parts = str(args.orig_name).split('/')
+orig_tutorial_name = orig_name_parts[0]
+new_tutorial_name_parts = str(args.new_name).split('/')
+new_tutorial_name = new_tutorial_name_parts[0]
+orig_tutorial_type = ''
+new_tutorial_type = ''
+if len(orig_name_parts) > 1:
+    orig_tutorial_type = orig_name_parts[1]
+if len(new_tutorial_name_parts) > 1:
+    new_tutorial_type = new_tutorial_name_parts[1]
+if args.debug:
+    print("Names of tutorial to be cloned " + str(orig_name_parts) + '  ' + orig_tutorial_name
+          + '  ' + str(orig_tutorial_type))
+proposed_tutorial = tutorials_dir + '/' + new_tutorial_name + '/' + new_tutorial_type
+original_tutorial = tutorials_dir + '/' + orig_tutorial_name + '/' + orig_tutorial_type
+
+if args.debug:
+    print('Original tutorial ' + original_tutorial)
+    print('Proposed tutorial ' + proposed_tutorial)
+
+print("Validating the tutorial name(s)")
+ensure_valid_tutorial_names()
+
+print("Copying the original files from {old} to \n {new}".format(old=original_tutorial, new=proposed_tutorial))
+print("This will also create all required directories")
+shutil.copytree(original_tutorial, proposed_tutorial)
+
+print("Copy the needed YAML files to %s" % test_harness_dir + '/' + new_tutorial_name)
+print("Then do the required replacements for getting the correct paths")
+types_to_enable = handle_harness_files(orig_tutorial_type, new_tutorial_type,
+                                       test_harness_dir, orig_tutorial_name, new_tutorial_name)
+
+print("Updating copied files from {old} to {new}".format(old=args.orig_name, new=args.new_name))
+do_replacements(proposed_tutorial, args.orig_name, args.new_name)
+
+if args.orig_main_class is not None and args.new_main_class is not None:
+    print("Renaming files from {orig} to {new}".format(orig=args.orig_main_class, new=args.new_main_class))
+    do_replacements(proposed_tutorial, args.orig_main_class, args.new_main_class)
+    find_and_rename_files_with_class_names(proposed_tutorial, args.orig_main_class, args.new_main_class)
+
+print("Now updating the semaphore.yml file with the added test")
+yaml_obj = get_yaml_file(semaphore_path)
+new_job = create_semaphore_job(args.semaphore_test_name, args.new_name)
+yaml_obj = update_semaphore_yaml(yaml_obj, get_semaphore_test_block(new_tutorial_type), new_job)
+write_yaml_to_file(semaphore_path, yaml_obj)
+
+print("Updating the tutorials.yml file")
+tutorials_yaml = get_yaml_file(tutorials_yaml_path)
+tutorials_yaml = update_tutorials_yaml(tutorials_yaml, new_tutorial_name, types_to_enable)
+write_yaml_to_file(tutorials_yaml_path, tutorials_yaml)
+print("Now updating front matter entries")
+if args.permalink is not None:
+    update_front_matter(orig_tutorial_name, new_tutorial_name, orig_tutorial_type, new_tutorial_type, args.permalink)
+else:
+    update_front_matter(orig_tutorial_name, new_tutorial_name, orig_tutorial_type, new_tutorial_type)
```

Comment:
collapse to one line since permalink arg is optional with default `None`?  i.e.
```
update_front_matter(orig_tutorial_name, new_tutorial_name, orig_tutorial_type, new_tutorial_type, args.permalink)
```